### PR TITLE
Add content list

### PR DESCRIPTION
**Pull request by itkg-nanne**
https://github.com/open-orchestra/open-orchestra-model-bundle/pull/722
https://github.com/open-orchestra/open-orchestra-model-interface/pull/280
https://github.com/open-orchestra/open-orchestra-mongo-libs/pull/66
https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/2068
https://github.com/open-orchestra/open-orchestra-libs/pull/74
https://github.com/open-orchestra/open-orchestra/pull/1131

### CODE REVIEW

File: ApiBundle/Controller/ContentController.php [hunk 1]
```diff
@@ -29,144 +26,62 @@
...
-     * @Config\Method({"GET"})
+     * @Api\Groups({
+     *     OpenOrchestra\ApiBundle\Context\CMSGroupContext::FIELD_TYPES
+     * })
      *
      * @return FacadeInterface
-     * @throws SourceLanguageNotFoundHttpException
      */
-    public function showOrCreateAction(Request $request, $contentId)
+    public function listAction(Request $request, $contentTypeId, $siteId, $language)
```
Comment (alavieille) [inline ApiBundle/Controller/ContentController.php]: phpdoc

File: ApiBundle/Controller/ContentController.php [hunk 2]
```diff
@@ -29,144 +26,62 @@
...
-        $siteId = $this->get('open_orchestra_backoffice.context_manager')->getCurrentSiteId();
-
-        $repository =  $this->get('open_orchestra_model.repository.content');
-        $transformer = $this->get('open_orchestra_api.transformer_manager')->get('content_collection');
-
-        if ($request->get('entityId') && $request->get('language')) {
-            $content = $this->showOrCreate($request, $request->get('entityId'));
-
-            return $transformer->transform(array($content), $contentType);
+        foreach ($contentType->getFields() as $field) {
```
Comment (alavieille) [inline ApiBundle/Controller/ContentController.php]: why not use MappingContentAttribute ?

File: ApiBundle/Controller/ContentController.php [hunk 3]
```diff
@@ -29,144 +26,62 @@
...
 {
     use ListStatus;
-    use HandleRequestDataTable;
 
     /**
      * @param Request $request
-     * @param string  $contentId
      *
-     * @Config\Route("/{contentId}", name="open_orchestra_api_content_show")
+     * @Config\Route("/list/{contentTypeId}/{siteId}/{language}", name="open_orchestra_api_content_list")
```
Comment (alavieille) [inline ApiBundle/Controller/ContentController.php]: update function test

File: ApiBundle/Controller/ContentController.php [hunk 4]
```diff
@@ -188,16 +103,25 @@ public function newVersionAction(Request $request, $contentId)
     }
 
     /**
-     * @param string  $contentId
+     * @param Request $request
      *
-     * @Config\Route("/{contentId}/duplicate", name="open_orchestra_api_content_duplicate")
+     * @Config\Route("/duplicate", name="open_orchestra_api_content_duplicate")
```
Comment (alavieille) [inline ApiBundle/Controller/ContentController.php]: update functional test

File: ApiBundle/Controller/ContentController.php [hunk 5]
```diff
@@ -213,6 +137,38 @@ public function duplicateAction($contentId)
 
     /**
      * @param Request $request
+     *
+     * @Config\Route("/delete-multiple", name="open_orchestra_api_content_delete_multiple")
```
Comment (alavieille) [inline ApiBundle/Controller/ContentController.php]: add functional test

File: ApiBundle/Transformer/FieldTypeTransformer.php [hunk 6]
```diff
@@ -19,10 +19,14 @@ class FieldTypeTransformer extends AbstractTransformer
      * @param string                               $facadeClass
      * @param MultiLanguagesChoiceManagerInterface $multiLanguagesChoiceManagerInterface
      */
-    public function __construct($facadeClass, MultiLanguagesChoiceManagerInterface $multiLanguagesChoiceManagerInterface)
-    {
+    public function __construct(
```
Comment (alavieille) [inline ApiBundle/Transformer/FieldTypeTransformer.php]: phpdoc

File: BackofficeBundle/Resources/public/ecmascript/OpenOrchestra/Application/Router/Content/ContentRouter.js [hunk 7]
```diff
@@ -50,6 +55,61 @@ class ContentRouter extends OrchestraRouter
...
+            siteId: Application.getContext().siteId,
+            language: language,
+            contentTypeName: contentTypeName,
+        };
+        new ContentType().fetch({
+            urlParameter: {
+                contentTypeId: contentTypeId
+            },
+            success: (contentType) => {
+                new Statuses().fetch({
```
Comment (alavieille) [inline BackofficeBundle/Resources/public/ecmascript/OpenOrchestra/Application/Router/Content/ContentRouter.js]: not use nested xhr request, use promise

File: BackofficeBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Content/ContentListView.js [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+        };
+        for (let column in defaultListable) {
+            if (defaultListable[column]) {
+                columnsDefinition.push({
+                    name: column,
+                    title: Translator.trans('open_orchestra_backoffice.table.contents.' + column),
+                    orderable: true,
+                    activateColvis: true,
+                    visible: true,
+                    createdCell: createdCell.hasOwnProperty(column) ? createdCell[column].bind(this) : void(0)
```
Comment (alavieille) [inline BackofficeBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Content/ContentListView.js]: i don't undersant this line

File: BackofficeBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Content/ContentsView.js [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+/**
+ * @class ContentsView
+ */
+class ContentsView extends AbstractCollectionView
+{
+    /**
+     * @inheritdoc
+     */
+    initialize({collection, settings, urlParameter, contentType, statuses}) {
+        this._collection = collection;
```
Comment (alavieille) [inline BackofficeBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Content/ContentsView.js]: you should call super

File: BackofficeBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Content/ContentsView.js [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+        statuses = statuses.hasOwnProperty('statuses') ? statuses.statuses : [];
+        let template = this._renderTemplate('Content/contentsView', {
+            contentType: this._contentType.toJSON(),
+            language: this._urlParameter.language,
+            siteLanguages: Application.getContext().siteLanguages,
+            statuses: statuses,
+            SearchFormGroupManager: SearchFormGroupManager
+        });
+        this.$el.html(template);
+        $.datepicker.setDefaults($.datepicker.regional[Application.getContext().language]);
```
Comment (alavieille) [inline BackofficeBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Content/ContentsView.js]: i think isn't a good idea to update default local for all datepicker

File: BackofficeBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Content/ContentsView.js [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+        $('.contents-list', this.$el).html(this._listView.render().$el);
+
+        return this;
+    }
+    /**
+     * Remove
+     *
+     * @private
+     */
+    _remove() {
```
Comment (alavieille) [inline BackofficeBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Content/ContentsView.js]: this method is already defined in abstract with the same code

File: BackofficeBundle/Resources/public/ecmascript/OpenOrchestra/Service/Content/DateSearchFormGroup.js [hunk 12]
```diff
@@ -0,0 +1,30 @@
+import TemplateManager from '../TemplateManager'
+
+/**
+ * @class DateSearchFormGroup
+ */
+class DateSearchFormGroup
```
Comment (alavieille) [inline BackofficeBundle/Resources/public/ecmascript/OpenOrchestra/Service/Content/DateSearchFormGroup.js]: extend AbstractSearchFormGroup no ?

File: BackofficeBundle/Resources/public/ecmascript/OpenOrchestra/Service/Content/NumberSearchFormGroup.js [hunk 13]
```diff
@@ -0,0 +1,30 @@
+import TemplateManager from '../TemplateManager'
+
+/**
+ * @class NumberSearchFormGroup
+ */
+class NumberSearchFormGroup
```
Comment (alavieille) [inline BackofficeBundle/Resources/public/ecmascript/OpenOrchestra/Service/Content/NumberSearchFormGroup.js]: same

File: BackofficeBundle/Resources/public/ecmascript/OpenOrchestra/Service/Content/SearchFormGroupManager.js [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+     */
+    constructor() {
+        this._fieldSearchs = [];
+    }
+
+    /**
+     * @param {Object} fieldSearch
+     */
+    add(fieldSearch) {
+        this._fieldSearchs.push(fieldSearch);
```
Comment (alavieille) [inline BackofficeBundle/Resources/public/ecmascript/OpenOrchestra/Service/Content/SearchFormGroupManager.js]: check extend AbstractSearchFormGroup

File: BackofficeBundle/Resources/public/ecmascript/OpenOrchestra/Service/Content/TextSearchFormGroup.js [hunk 15]
```diff
@@ -0,0 +1,30 @@
+import TemplateManager from '../TemplateManager'
+
+/**
+ * @class TextSearchFormGroup
+ */
+class TextSearchFormGroup
```
Comment (alavieille) [inline BackofficeBundle/Resources/public/ecmascript/OpenOrchestra/Service/Content/TextSearchFormGroup.js]: extend AbstractSearchFormGroup

File: BackofficeBundle/Resources/public/ecmascript/OpenOrchestra/Service/Form/Behavior/DatePicker.js [hunk 16]
```diff
@@ -38,6 +39,7 @@ class DatePicker extends AbstractBehavior
             dataDateFormat = this._convertFormat(this._convertFormatYear, dataDateFormat);
             dataDateFormat = this._convertFormat(this._convertFormatMonth, dataDateFormat);
             dataDateFormat = this._convertFormat(this._convertFormatDay, dataDateFormat);
+            $.datepicker.setDefaults($.datepicker.regional[Application.getContext().language]);
```
Comment (alavieille) [inline BackofficeBundle/Resources/public/ecmascript/OpenOrchestra/Service/Form/Behavior/DatePicker.js]: same i think isn't a ggod idea to force for all datepicker

File: BackofficeBundle/Resources/translations/interface.en.yml [hunk 17]
```diff
@@ -42,6 +42,9 @@ open_orchestra_backoffice:
           site: Sites
     table:
         delete_button: Remove selected items
+        duplicate: Duplicate
+        'true': Yes
```
Comment (alavieille) [inline BackofficeBundle/Resources/translations/interface.en.yml]: indent

File: BackofficeBundle/Resources/translations/interface.en.yml [hunk 18]
```diff
@@ -121,3 +131,8 @@ open_orchestra_backoffice:
         swchoff:
             on: Yes
             off: No
+    language:
+        en: English
+        fr: Français
+        es: Español
+        de: Deutsch
```
Comment (alavieille) [inline BackofficeBundle/Resources/translations/interface.en.yml]: eof

File: GroupBundle/Controller/Api/GroupController.php [hunk 19]
```diff
@@ -90,9 +90,16 @@ public function deleteGroupsAction(Request $request)
...
+        $filter = $groups;
+        array_walk($filter, function(&$item) {$item = $item->getId();});
+        $nbrGroupsUsers = $this->get('open_orchestra_user.repository.user')->getCountsUsersByGroups($filter);
+
         $groupIds = array();
         foreach ($groups as $group) {
-            if ($this->isGranted(ContributionActionInterface::DELETE, $group)) {
+            if ($this->isGranted(ContributionActionInterface::DELETE, GroupInterface::ENTITY_TYPE) &&
+                array_key_exists($group->getId(), $nbrGroupsUsers) &&
+                $nbrGroupsUsers[$group->getId()] == 0) {
```
Comment (alavieille) [inline GroupBundle/Controller/Api/GroupController.php]: yoda notation

File: BackofficeBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Content/ContentsView.js (hunk 10 above)
Comment (itkg-nanne) [inline BackofficeBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Content/ContentsView.js]: Could not be done in another way.

**Comment by itkg-nanne** [comment]
update doen @alavieille

File: BackofficeBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Content/ContentListView.js [hunk 20]
```diff
@@ -46,7 +47,7 @@ class ContentListView extends mix(AbstractDataTableView).with(UrlPaginateViewMix
         let columnsDefinition = [];
         let defaultListable = this._contentType.get('default_listable');
         let createdCell = {
-            'linked_to_site': this._createBoolean,
+            'linked_to_site': 'bool',
```
Comment (alavieille) [inline BackofficeBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Content/ContentListView.js]: you can use a const ?

**Comment by itkg-nanne** [comment]
update done @alavieille

**Pull request merged**